crates/extrinsics/src/call.rs

```diff
@@ -90,12 +87,123 @@ pub struct CallCommand {
+            },
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<M> CallCommandBuilder<M, Missing<state::ExtrinsicOptions>> {
+    /// Sets the extrinsic operation.
+    pub fn extrinsic_opts(
+        self,
+        extrinsic_opts: ExtrinsicOpts,
+    ) -> CallCommandBuilder<M, state::ExtrinsicOptions> {
+        CallCommandBuilder {
+            opts: CallCommand {
+                extrinsic_opts,
+                ..self.opts
+            },
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<M, E> CallCommandBuilder<M, E> {
+    /// Sets the the address of the the contract to call.
+    pub fn contract(self, contract: <DefaultConfig as Config>::AccountId) -> Self {
+        let mut this = self;
+        this.opts.contract = contract;
+        this
+    }
+
+    /// Sets the arguments of the contract message to call.
+    pub fn args(self, args: Vec<String>) -> Self {
+        let mut this = self;
+        this.opts.args = args;
+        this
+    }
+
+    /// Sets the maximum amount of gas to be used for this command.
+    pub fn gas_limit(self, gas_limit: u64) -> Self {
+        let mut this = self;
+        this.opts.gas_limit = Some(gas_limit);
+        this
+    }
+
+    /// Sets the maximum proof size for this call.
+    pub fn proof_size(self, proof_size: u64) -> Self {
+        let mut this = self;
+        this.opts.proof_size = Some(proof_size);
+        this
+    }
+
+    /// Sets the value to be transferred as part of the call.
+    pub fn value(self, value: BalanceVariant) -> Self {
+        let mut this = self;
+        this.opts.value = value;
+        this
+    }
+
+    /// Sets whether to export the call output in JSON format.
+    pub fn output_json(self, output_json: bool) -> Self {
+        let mut this = self;
+        this.opts.output_json = output_json;
+        this
+    }
+}
+
+impl CallCommandBuilder<state::Message, state::ExtrinsicOptions> {
+    /// Finishes construction of the call command.
+    pub async fn done(self) -> CallExec {
+        let call_command = self.opts;
+        call_command.preprocess().await.unwrap()
+    }
+}
+
+#[allow(clippy::new_ret_no_self)]
 impl CallCommand {
+    /// Returns a clean builder for [`CallCommand`].
+    pub fn new(
+    ) -> CallCommandBuilder<Missing<state::Message>, Missing<state::ExtrinsicOptions>>
+    {
+        CallCommandBuilder {
+            opts: Self {
+                contract: AccountId32([0; 32]),
+                message: String::new(),
+                args: Vec::new(),
+                extrinsic_opts: ExtrinsicOpts::default(),
+                gas_limit: None,
+                proof_size: None,
+                value: "0".parse().unwrap(),
+                output_json: false,
+            },
+            marker: PhantomData,
+        }
+    }
+
     pub fn is_json(&self) -> bool {
         self.output_json
     }
 
-    pub fn run(&self) -> Result<(), ErrorVariant> {
+    /// Helper method for preprocessing contract artifacts.
```

Comment:
Just by the method name it's not clear when and how this should be used. Can you please explain it better in the doc comment?